pkgs/applications/networking/instant-messengers/feishu/default.nix

```diff
@@ -57,19 +57,51 @@
 , wayland
 , wrapGAppsHook3
 , xdg-utils
+, writeScript
 
-# for custom command line arguments, e.g. "--use-gl=desktop"
+  # for custom command line arguments, e.g. "--use-gl=desktop"
 , commandLineArgs ? ""
 }:
 
 stdenv.mkDerivation rec {
-  version = "6.9.20";
+  version = "7.15.13";
   pname = "feishu";
-  packageHash = "6085d1c4"; # A hash value used in the download url
 
-  src = fetchurl {
-    url = "https://sf3-cn.feishucdn.com/obj/ee-appcenter/${packageHash}/Feishu-linux_x64-${version}.deb";
-    hash = "sha256-kg5j/vWaCBUjnF983kk0ZMJ+inF5z5ctED9+ITuIn94=";
+  src = passthru.sources.${stdenv.hostPlatform.system};
```

Comment:
If we really need `passthru.sources`, it should be `passthru` inheriting from `src`, rather than the other way. Also we need to throw error if the user's arch isn't supported.

I just found an example for your reference: https://github.com/NixOS/nixpkgs/blob/69aa70cddf9d8c7b535c0f456a5574f5e85c4271/pkgs/servers/sql/cockroachdb/cockroachdb-bin.nix#L17

Reply:
Thanks, fixed.